### PR TITLE
Add Apache Polaris converter module

### PR DESCRIPTION
**Pull request by jbonofre**
## Summary

- Adds a two-way converter between OSI semantic models and Apache Polaris catalogs under `converters/polaris/`
- **Import mode**: reads Polaris namespaces/tables via the Iceberg REST Catalog API and generates OSI YAML, mapping Iceberg schemas to OSI datasets, fields, primary keys, and temporal dimensions
- **Export mode**: reads OSI YAML and creates Polaris namespaces and Iceberg tables with type inference from field names and dimension metadata

## Test plan

- [x] Unit tests pass (9 tests covering parsing, YAML generation round-trip, Iceberg schema generation, type inference, and edge cases)
- [ ] Integration test against a running Polaris instance (import and export)
- [ ] Verify round-trip fidelity: Polaris → OSI → Polaris

### CODE REVIEW

**Comment by jbonofre** [comment]
This is a use example:

```
  # Import: Polaris catalog → OSI YAML
  java -jar osi-polaris-converter.jar import \
    --url http://localhost:8181 --catalog my_catalog \                                                                                                                                                                                        
    --client-id <id> --client-secret <secret> \
    -o output.yaml                                                                                                                                                                                                                            
                                                            
  # Export: OSI YAML → Polaris catalog                                                                                                                                                                                                        
  java -jar osi-polaris-converter.jar export \              
    --url http://localhost:8181 --catalog my_catalog \
    --token <bearer-token> \                                                                                                                                                                                                                  
    model.yaml
```
                                                                                                                                                                                                                                              
 I implemented with the following logic:
 * Each Polaris namespace maps to an OSI semantic model                                                                                                                                                                                      
 * Each Iceberg table maps to an OSI dataset               
 * Iceberg schema fields map to OSI fields with ANSI_SQL expressions
 * Iceberg identifier-field-ids map to OSI primary_key                                                                                                                                                                                       
 * Temporal Iceberg types (timestamp, timestamptz, date) set dimension.is_time: true                                                                                                                                                         
 * Table properties are preserved in custom_extensions

File: converters/polaris/src/main/java/org/osi/converter/polaris/model/OsiModel.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+package org.osi.converter.polaris.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Java representation of an OSI semantic model parsed from YAML.
+ */
+public class OsiModel {
```
Comment (flyrain) [inline converters/polaris/src/main/java/org/osi/converter/polaris/model/OsiModel.java:9]: Thanks for adding this, @jbonofre.  Can we put it a common module instead of Polaris module? I think the  Java representation would be really useful for other catalogs as well.

Comment (jbonofre) [reply converters/polaris/src/main/java/org/osi/converter/polaris/model/OsiModel.java:9]: @flyrain That's a good idea. I think it also related to a discussion we had before: what should be the "language layout" as we have converter in Python, Java, ...

Let me give a proposal.

Comment (flyrain) [reply converters/polaris/src/main/java/org/osi/converter/polaris/model/OsiModel.java:9]: Awesome! Looking forward to it

Comment (jbonofre) [reply converters/polaris/src/main/java/org/osi/converter/polaris/model/OsiModel.java:9]: I'm working on a PR to have `core-spec` module, that will create bindings for different languages (including java).

**Comment by jbonofre** [comment]
@khush-bhatia I'm preparing a first step PR with `core-spec` and java binding.

**Review by khush-bhatia — APPROVED** [review]

**Review by sgandon — APPROVED** [review]
LGTM

**Comment by jbonofre** [comment]
I created #121 for follow up.
